### ISSUE TITLE
Please fix warnings and errors in examples

### ISSUE THREAD

**Issue by yurivict** [kcat/openal-soft#133]
Some of errors:
```
/usr/local/include/libavcodec/avcodec.h:3977:14: note: 'data' has been explicitly marked deprecated here
    uint8_t *data[AV_NUM_DATA_POINTERS];    ///< pointers to the image data planes
             ^
examples/alffplay.c:945:32: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        if(movState->video.st->codec->pix_fmt == AV_PIX_FMT_YUV420P)
                               ^
examples/alffplay.c:916:41: warning: 'codec' is deprecated [-Wdeprecated-declarations]
        vp->width = movState->video.st->codec->width;
                                        ^
```

also this change is needed:
```
--- examples/alffplay.cpp.orig  2017-07-23 04:34:58 UTC
+++ examples/alffplay.cpp
@@ -16,6 +16,7 @@
 #include <atomic>
 #include <mutex>
 #include <deque>
+#include <array>
 
 extern "C" {
 #include "libavcodec/avcodec.h"
```

Version 1.17.2 on FreeBSD
clang-4.0

**Comment by yurivict** [comment]
18.0 also has these errors:
```
examples/alffplay.cpp:205:50: error: implicit instantiation of undefined template 'std::__1::array<(anonymous namespace)::VideoState::Picture, 16>'
    std::array<Picture,VIDEO_PICTURE_QUEUE_SIZE> mPictQ;
                                                 ^
/usr/include/c++/v1/__tuple:222:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
examples/alffplay.cpp:740:56: error: implicit instantiation of undefined template 'std::__1::array<unsigned int, 8>'
            std::array<ALuint,AUDIO_BUFFER_QUEUE_SIZE> tmp;
                                                       ^
/usr/include/c++/v1/__tuple:222:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;
                                                               ^
In file included from examples/alffplay.cpp:21:
/usr/local/include/libavcodec/avcodec.h:1634:16: warning: 'convergence_duration' is deprecated [-Wdeprecated-declarations]
typedef struct AVPacket {
               ^
/usr/include/c++/v1/memory:1694:18: note: in instantiation of function template specialization 'std::__1::allocator<AVPacket>::construct<AVPacket, AVPacket>' requested here
            {__a.construct(__p, _VSTD::forward<_Args>(__args)...);}
                 ^
/usr/include/c++/v1/memory:1540:14: note: in instantiation of function template specialization 'std::__1::allocator_traits<std::__1::allocator<AVPacket> >::__construct<AVPacket, AVPacket>' requested here
            {__construct(__has_construct<allocator_type, _Tp*, _Args...>(),
             ^
/usr/include/c++/v1/deque:1822:21: note: in instantiation of function template specialization 'std::__1::allocator_traits<std::__1::allocator<AVPacket> >::construct<AVPacket, AVPacket>' requested here
    __alloc_traits::construct(__a, _VSTD::addressof(*__base::end()), _VSTD::move(__v));
                    ^
examples/alffplay.cpp:288:14: note: in instantiation of member function 'std::__1::deque<AVPacket, std::__1::allocator<AVPacket> >::push_back' requested here
    mPackets.push_back(AVPacket{});
             ^
/usr/local/include/libavcodec/avcodec.h:1686:13: note: 'convergence_duration' has been explicitly marked deprecated here
    int64_t convergence_duration;
            ^
```

**Comment by kcat** [comment]
> Version 1.17.2 on FreeBSD

These are fixed in 1.18.0.

> 18.0 also has these errors:

The errors should be fixed in commit 4ec67e6226e956d8f87c67976d3c6b07c8d2fb2c.

**Comment by yurivict** [comment]
Ok, thanks!
